packages/fluentui/react-northstar-prototypes/src/prototypes/VirtualizedStickyTree/VirtualStickyTree.tsx

```diff
@@ -0,0 +1,368 @@
+            style, // came from react-window
+            expanded,
+            parent,
+            key: id,
+            level,
+            index,
+            treeSize,
+            selectable: false,
+            onFocus: () => makeVisibleOnFocus(id, level),
+            ...(level === 1 && expanded && childrenIds.length && { onKeyDown: handleArrowUpDownOnSticky(id, item) }),
+          },
+        });
+      }
+      return null;
+    },
+    [getA11yProps, getItemById, handleArrowUpDownOnSticky, makeVisibleOnFocus, props.renderItemTitle],
+  );
+
+  const innerElementContextValue: InnerElementContextType = React.useMemo(
+    () => ({
+      getItemById,
+      stickyItemIds,
+      stickyItemPusherHeights,
+      stickyItemSize,
+    }),
+    [getItemById, stickyItemIds, stickyItemPusherHeights, stickyItemSize],
+  );
+
+  const getItemKey = React.useCallback((index: number, data: VirtualItemData) => data.visibleItemIds[index], []);
+
+  const getItemSize = React.useCallback(
+    (index: number) => {
+      const id = visibleItemIds[index];
+
+      const item = getItemById(id);
+      if (item?.level === 1) {
+        return stickyItemSize;
+      }
+      return itemSize;
+    },
+    [getItemById, itemSize, stickyItemSize, visibleItemIds],
+  );
+
+  React.useLayoutEffect(() => {
+    listRef.current.resetAfterIndex(0);
+  }, [listRef, visibleItemIds]); // when item collapsed/expanded (visibleItemIds change), refresh react-window itemSize cache
+
+  const element = (
+    <TreeContext.Provider value={contextValue}>
+      {getA11yProps.unstable_wrapWithFocusZone(
+        <ElementType
+          {...getA11yProps('root', {
+            className: classes.root,
+            ...rtlTextContainer.getAttributes({ forElements: [children] }),
+            ...unhandledProps,
+          })}
+        >
+          <InnerElementContext.Provider value={innerElementContextValue}>
+            <VariableSizeList
+              width={-1} // width is not used for vertical list
+              ref={listRef}
+              height={height}
+              itemSize={getItemSize}
+              itemKey={getItemKey}
+              itemData={{ visibleItemIds, createTreeItem }}
+              itemCount={visibleItemIds.length}
+              outerElementType={OuterElementType}
+              innerElementType={InnerElementType}
+            >
+              {ItemWrapper}
+            </VariableSizeList>
+          </InnerElementContext.Provider>
+        </ElementType>,
+      )}
+    </TreeContext.Provider>
+  );
+  return element;
+};
+
+const getStickyItemStyle = (indexAmoungStickyItems: number, stickyItemNums: number, stickyItemSize: number) => ({
+  height: stickyItemSize,
+  zIndex: teamsTheme.siteVariables.zIndexes.overlay,
+  position: 'sticky',
+  top: indexAmoungStickyItems * stickyItemSize,
+  bottom: (stickyItemNums - indexAmoungStickyItems - 1) * stickyItemSize,
+  backgroundColor: teamsTheme.siteVariables.colorScheme.default.background3,
```

Comment:
are we supposed to handle theme switching in prototypes ?

Reply:
nope I don't think we handle theme in prototypes at all.
This is just to give sticky header a different color.